### DISCUSSION TITLE
I am having trouble consuming Google OAuth using NextAuthJS with NextJS 14 and Prisma.

### DISCUSSION THREAD

**Discussion by devpedro-dias** [community/community#146816] [Programming Help]
#### I've checked the environment variables, practically did several things, and the error that appears to me in the console is this below:

I practically understood that the problem is possibly related to Prisma, because of the "findUnique", but I do not know what is wrong. If anyone can help me, I will be grateful!

Repo: https://github.com/devpedro-dias/cet-vtx3

![image](https://github.com/user-attachments/assets/445c2762-e78d-4e87-b2a0-4ecdedc78e8b)
![image](https://github.com/user-attachments/assets/22e6bb4e-50e3-42ca-b5e6-2d657f1de70a)

**Comment by richardfariax** [comment]
NextAuth + Google + Prisma usually breaks on one of these — the `findUnique` in the stack trace is often a symptom, not the root cause:

1. **Adapter schema** — `User`, `Account`, `Session`, `VerificationToken` must match `@auth/prisma-adapter`. Easiest fix: copy the official Auth.js Prisma schema and run `migrate dev`.

2. **PrismaClient singleton** — hot reload in dev can spawn multiple clients. Use a global singleton in `lib/prisma.ts`.

3. **Env vars** — `DATABASE_URL` reachable at runtime; Google redirect URI must match exactly (`/api/auth/callback/google` for Auth.js v5).

4. **Version mismatch** — Auth.js v5 uses `AUTH_SECRET`, v4 uses `NEXTAUTH_SECRET`. Mixing docs from both causes weird adapter errors.

Paste the full stack trace + your `Account`/`User` models if it's still failing — hard to tell without the exact `findUnique` query.
